Add the `take_over` mode for filestream inputs (#34292)
elastic/beats: filebeat/docs/howto/migrate-to-filestream.asciidoc

[[migrate-to-filestream]]
== Migrate `log` input configurations to `filestream`

> The `filestream` input has been generally available since 7.14. So, it is highly recommended

Reviewer: ```suggestion
The `filestream` input has been generally available since 7.14 and it is highly recommended
```

The `So, ` still makes it seem informal which is a bit weird to me in technical documentation. You removed the "high time" already but I think we can go even further :)

you migrate your existing `log` input configurations. The `filestream` input comes with many
improvements over the old `log` input, such as configurable order for parsers and more.

> The `log` input is deprecated and will be eventually removed from Filebeat. We are not fixing

Reviewer: ```suggestion
The `log` input is deprecated and will eventually be removed from Filebeat. We are not fixing
```

new issues or adding any enhancements to the `log` input. Our focus is on `filestream`.

This manual migration is required only if you've defined `log` inputs manually
in your stand-alone Filebeat configuration. All the integrations or modules that are still using
`log` inputs under the hood will be eventually migrated automatically without any additional actions
required from the user.

In this guide, you'll learn how to migrate an existing `log` input configuration.

IMPORTANT: You must replace `log` inputs with `filestream` inputs, make sure you have removed

> the all old `log` inputs from the configuration before starting Filebeat with the new `filestream` inputs. Running old `log` inputs and new `filestream` inputs pointed to the same files will lead to data duplication.

Reviewer: ```suggestion
all the old `log` inputs from the configuration before starting Filebeat with the new `filestream` inputs. Running old `log` inputs and new `filestream` inputs pointed to the same files will lead to data duplication.
```

=== Step 1: Set an identifier for each `filestream` input

> All `filestream` inputs require an ID. So make sure you set a unique identifier for every input.

Reviewer: ```suggestion
All `filestream` inputs require an ID. Ensure you set a unique identifier for every input.
```

|backoff.max
|===

=== If something went wrong

If for whatever reason you'd like to revert the configuration after running the migrated configuration

> and return old `log` inputs their files taken by `filestream` inputs, you need to do the following:

Reviewer: ```suggestion
and return to old `log` inputs the files that were taken by `filestream` inputs, you need to do the following:
```
